apex/provider/resource_block_storage.go

```diff
@@ -574,45 +569,31 @@ func (r *blockStorageResource) Create(ctx context.Context, req resource.CreateRe
 		)
 		return
 	}
-	if err := status.Body.Close(); err != nil {
-		fmt.Print("Error Closing response body:", err)
-	}
 
 	// Fetching Job Status
-	poller := helper.NewPoller(r.jobsClient)
-	resourceID, err := poller.WaitForResource(ctx, job.Id)
-	if err != nil {
+	resourceID, jobErr := helper.JobStatus(ctx, r.jobsClient, job.Id)
+	if jobErr != nil {
 		resp.Diagnostics.AddError(
 			"Error getting resourceID",
-			helper.ResourceRetrieveError+err.Error(),
+			helper.ResourceRetrieveError+jobErr.Error(),
 		)
 		return
 	}
 
-	if err := status.Body.Close(); err != nil {
-		fmt.Print("Error Closing response body:", err)
-	}
-
 	// Fetching Block storage after Job Completes
-	storageSystem, status, err := r.client.StorageSystemsAPI.StorageSystemsInstance(ctx, resourceID).Execute()
+	storageSystem, status, err := helper.GetBlockStorageInstance(r.client, resourceID)
 	if (err != nil) || (status.StatusCode != http.StatusOK) {
-		if err := status.Body.Close(); err != nil {
-			fmt.Print("Error Closing response body:", err)
-		}
 		// Try again with POWERFLEX- prefix
 		if status.StatusCode == http.StatusNotFound {
 			resourceIDWithPrefix := "POWERFLEX-" + resourceID
-			storageSystem, status, err = r.client.StorageSystemsAPI.StorageSystemsInstance(ctx, resourceIDWithPrefix).Execute()
+			storageSystem, status, err = helper.GetBlockStorageInstance(r.client, resourceIDWithPrefix)
 			if (err != nil) || (status.StatusCode != http.StatusOK) {
 				resp.Diagnostics.AddError(
 					"Error retrieving created Block storage",
 					"Could not retrieve created Block storage, unexpected error: "+err.Error(),
```

Comment:
@doriac11 Here error could be nil. May be use GetErrorString() method from general helper to avoid nil pointer deference error to generate msg.

Reply:
Good point, will update this.